File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Request/MeterReadingCreateRequest.java
package ut.edu.be_quanlytro.Dto.Request;

import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import java.time.LocalDate;
import java.util.UUID;

@Data
public class MeterReadingCreateRequest {
    @NotNull
    private UUID roomId;
    @NotNull
    private UUID serviceId;
    @NotNull
    @Min(value = 0)
    private Integer newIndex;
    @NotNull
    private LocalDate readingDate;
}

File: Flutter_QuanLyTro/lib/data/models/request/meter_reading_create_request.dart
class MeterReadingCreateRequest {
  final String roomId;
  final String serviceId;
  final int newIndex;
  final String readingDate;

  MeterReadingCreateRequest({
    required this.roomId,
    required this.serviceId,
    required this.newIndex,
    required this.readingDate,
  });

  Map<String, dynamic> toJson() {
    return {
      'roomId': roomId,
      'serviceId': serviceId,
      'newIndex': newIndex,
      'readingDate': readingDate,
    };
  }
}